scripts/ci/forbidden-identifiers.js

```diff
@@ -34,68 +28,135 @@ const blocked_statements = [
   'from \\\'rxjs/Rx\\\'',
 ];
 
-// Retrieves all scope packages dynamically from the source.
-// Developers shouldn't be able to import from other scope packages by using relative paths.
-const componentFolders = fs
-  .readdirSync(`${__dirname}/../../src/components`)
-  .map(componentName => `src/components/${componentName}`);
-
-const scopePackages = ['src/core'].concat(componentFolders);
-
+const sourceFolders = ['./src', './e2e'];
+const scopePackages = ['src/core'].concat(glob('src/components/*'));
 const blockedRegex = new RegExp(blocked_statements.join('|'), 'g');
 const importRegex = /from\s+'(.*)';/g;
 
-/**
- * Find the fork point between HEAD of the current branch, and master.
- * @return {Promise<string>} A promise which resolves with the fork SHA (or reject).
+/*
+ * Verify that the current PR is not adding any forbidden identifiers.
+ * Run the forbidden identifiers check against all sources when not verifying a PR.
  */
-function findForkPoint() {
-  return exec('git merge-base --fork-point HEAD master')
-    .then(function(stdout) {
-      return stdout.split('\n')[0];
+
+resolveFiles()
+
+  /* Only match .js or .ts, and remove .d.ts files. */
+  .then(fileList => {
+    return fileList.filter(name => name.match(/\.(js|ts)$/) && !name.match(/\.d\.ts$/));
```

Comment:
The most of those changes weren't actually from me. Also it seems like `.search` is not super elegant, because it returns `-1`, which can't be easily tested with a falsy check.

I recommend `regex.test()` which returns a plain boolean.